### ISSUE TITLE
Variable referencing itself is hard to detect.

### ISSUE THREAD

**Issue by DeputyLOL** [Mihara/RasterPropMonitor#586]
I tend to make changes to my cockpit without testing them in the actual game, because it takes along time to load or reload the database. 
The issue with that is I often forget I've made any changes at all.

I recently discovered that neither KSP or RPM are capable of handling a variable that references itself. The game just closes the instant you load an IVA that uses it. The logs aren't really that helpful as the last thing in the log is an rpm initialisation error relating to the animator that uses the bad variable. 

Long story short. If you mess up the variables and have one referencing itself, it is near impossible to find what and where the problem is as KSP won't stay open and RPM doesn't give any information in the log.

**Comment by MOARdV** [comment]
Yes, yes it is.  It is a programming hazard in most every language.  Circular reference detection would require parsing and "understanding" all of the custom variables by generating a graph of all of the math / select / custom variables in use, and creating a tree of all other custom variables each one references,  There is a non-trivial cost to adding that capability, and it's not something I would want enabled by default.  As a load-time test using an RPM config setting (`ValidateCustomVars = true`), perhaps something could be done.
